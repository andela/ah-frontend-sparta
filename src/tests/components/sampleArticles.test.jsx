import React from 'react';
import renderer from 'react-test-renderer';
import SampleArticles from '../../components/Articles';

describe('Sample Articles Component', () => {
  let mountApp;
  const props = {
    articles: [
      {
        id: 1,
        author: {
          username: 'rogha',
        },
        article_read_time: 'less than a minute read',
        title: 'Post one',
        description: 'too long a story',
        slug: 'post-one-first',
        body: 'ahhhhh --edit me',
        createdAt: '2019-04-18T11:55:58.111843Z',
      },
    ],
  };
  beforeEach(() => {
    mountApp = renderer.create(<SampleArticles {...props} />);
  });

  it('Should render without crashing', () => {
    expect(mountApp).toMatchSnapshot();
  });
});
