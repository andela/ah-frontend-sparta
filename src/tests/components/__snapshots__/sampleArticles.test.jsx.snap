// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Articles Component Should render without crashing 1`] = `
<div>
  <div>
    <div
      className="sample-articles"
    >
      <div
        className="article-image"
      >
        <img
          alt="Article"
          src={Object {}}
        />
      </div>
      <div
        className="trending-articles"
      >
        <h4>
          Post one
        </h4>
        <h6
          className="sample_article_start"
        >
          too long a story
        </h6>
        <p
          className="author_name"
        >
          Created By
           
          <b>
            rogha
          </b>
        </p>
        <h6>
          April 18, 2019
           
          -
          less than a minute read
        </h6>
      </div>
    </div>
  </div>
</div>
`;
